apps/nextjs-website/src/helpers/metadata.helpers.ts

```diff
@@ -1,27 +1,48 @@
 import { translations } from '@/_contents/translations';
-import { ResolvedMetadata } from 'next';
-import { AbsoluteTemplateString } from 'next/dist/lib/metadata/types/metadata-types';
-import { Twitter } from 'next/dist/lib/metadata/types/twitter-types';
+import { MakeMetadataFunction } from '@/lib/types/makeMetadataFunction';
+import { Metadata } from 'next';
 
-export const getTitleFromMarkdown = (markdown: string): string => {
-  const lines = markdown.split('\n').filter((line) => line !== '');
-  const titles = lines.filter((line) => line.startsWith('#'));
-  const title = titles[0]?.replace('# ', '').trim() ?? '';
-  return title;
+export const makeMetadata: MakeMetadataFunction = ({
+  parent,
+  title,
+  description,
+  url,
+  image,
+}) => {
+  const { shared } = translations;
+  const previousTitle = parent?.title?.absolute || shared.siteTitle;
+  const metadataTitle = title ? `${previousTitle} | ${title}` : previousTitle;
+
+  return {
+    title: metadataTitle,
+    description: description || '',
+    url: url || '',
+    openGraph: getOpenGraphMetadata(metadataTitle, description, image),
+    twitter: getTwitterMetadata(metadataTitle, description, image),
+  };
 };
 
-export const getTwitterMetadata = (title: string): Twitter => ({
+const getOpenGraphMetadata = (
+  title: string,
+  description?: string,
+  image?: string
+): Metadata['openGraph'] => ({
+  title,
+  type: 'website',
+  locale: 'it_IT',
```

Comment:
At the moment the i18n isn't really done, so I'd wait for that to complete first and the pass the locale as function parameter then. Also because I don't already know how the locale will be provided to the pages.

Reply:
Yeah, I get it, you could still pass the parameter and setting a default value, then when the i18n will be merged into `main`, you could simply update che caller manipulating the value of the locale and remove the default here in the helper function.

Anyway, it's ok also to wait.

Reply:
No worries, I updated the parameters